### PR TITLE
add MqttMessageFactory and optimize codes.

### PR DESCRIPTION
**Pull request by wangzhiwei1314**
add MqttMessageFactory and optimize codes.

### CODE REVIEW

**Comment by DongyuanPan** [comment]
Please open a new issue to explain why you want to do this, and associate this PR with the issue

File: mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
+package org.apache.rocketmq.mqtt.cs.protocol.mqtt.facotry;
```
Comment (ShannonDing) [inline mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java:1]: lack of license header.

File: mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.mqtt.*;
```
Comment (ShannonDing) [inline mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java:5]: import what used in detail.

**Review by DongyuanPan — APPROVED** [review]

**Comment by wangzhiwei1314** [comment]
Thanks for your review. Now I have fixed these problems.

File: mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java (hunk 1 above)
Comment (wangzhiwei1314) [reply mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java:19]: > lack of license header.

Thanks,I have fixed this problem.

File: mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java (hunk 2 above)
Comment (wangzhiwei1314) [reply mqtt-cs/src/main/java/org/apache/rocketmq/mqtt/cs/protocol/mqtt/facotry/MqttMessageFactory.java]: Thanks,I have fixed this problem.

**Comment by DongyuanPan** [comment]
please fix CI

**Comment by wangzhiwei1314** [comment]
> please fix CI

I fixed CI errors.

**Comment by DongyuanPan** [comment]
LGTM
@ShannonDing

**Review by ShannonDing — APPROVED** [review]

**Review by caigy — APPROVED** [review]
LGTM

**Review by DongyuanPan — APPROVED** [review]

**Review by pingww — APPROVED** [review]

**Pull request merged**